lib/ruby_smb/client/negotiation.rb

```diff
@@ -249,9 +249,11 @@ def smb2_3_negotiate_request
       def add_smb3_to_negotiate_request(packet, dialects = SMB3_DIALECT_DEFAULT)
         dialects.each do |dialect|
           raise ArgumentError, 'Must be an array of strings' unless dialect.is_a? String
+          next if dialect == '0x0311' && !@session_encrypt_data
```

Comment:
Let's update ruby_smb's constructor to take in the dialects as configurable option instead of this approach

Reply:
This wasn't added to the constructor, instead it was overridden directly in framework